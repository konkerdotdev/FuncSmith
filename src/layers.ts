/* eslint-disable @typescript-eslint/naming-convention */
import * as P from '@konker.dev/effect-ts-prelude';
import { MemFsTinyFileSystem, NodeTinyFileSystem } from '@konker.dev/tiny-filesystem-fp';

import { EMPTY_FILESET, FuncSmithContextSink, FuncSmithContextSource } from './index';
import type { FileSet, FileSetItem } from './lib/fileSet';
import { fsFileSinkWriter } from './sink/FsFileSinkWriter';
import { fsFileSourceReader } from './source/FsFileSourceReader';
import memFs1 from './test/fixtures/memfs-1.json';
import {
  FuncSmithContext,
  FuncSmithContextEnv,
  FuncSmithContextFs,
  FuncSmithContextMetadata,
  FuncSmithContextReader,
  FuncSmithContextWriter,
} from './types';

// --------------------------------------------------------------------------
export const FuncSmithContextFsLive = P.Layer.succeed(
  FuncSmithContextFs,
  FuncSmithContextFs.of({
    tinyFs: NodeTinyFileSystem,
  })
);
export const FuncSmithContextFsTest = P.Layer.succeed(
  FuncSmithContextFs,
  FuncSmithContextFs.of({
    tinyFs: MemFsTinyFileSystem(memFs1, '/tmp'),
  })
);

// --------------------------------------------------------------------------
export const FuncSmithContextSourceTest = FuncSmithContextSource.of({
  sourcePath: '/tmp/foo',
});

export const FuncSmithContextSinkTest = FuncSmithContextSink.of({
  sinkPath: '/tmp/build',
});

// --------------------------------------------------------------------------
export const FuncSmithContextReaderLayer = P.Layer.effect(
  FuncSmithContextReader,
  P.Effect.map(FuncSmithContextFs, (funcSmithContextFs) =>
    FuncSmithContextReader.of({
      tinyFs: funcSmithContextFs.tinyFs,
      reader: (sourcePath: string, globPattern: string | undefined) =>
        P.pipe(
          fsFileSourceReader(sourcePath, globPattern),
          P.Effect.provideService(FuncSmithContextFs, funcSmithContextFs)
        ),
    })
  )
);
export const FuncSmithContextReaderLive = P.pipe(FuncSmithContextReaderLayer, P.Layer.provide(FuncSmithContextFsLive));
export const FuncSmithContextReaderTest = P.pipe(FuncSmithContextReaderLayer, P.Layer.provide(FuncSmithContextFsTest));

// --------------------------------------------------------------------------
export const FuncSmithContextWriterLayer = P.Layer.effect(
  FuncSmithContextWriter,
  P.Effect.map(FuncSmithContextFs, (funcSmithContextFs) =>
    FuncSmithContextWriter.of({
      tinyFs: funcSmithContextFs.tinyFs,
      writer: (sinkPath: string, fileSet: FileSet<FileSetItem>) =>
        P.pipe(fsFileSinkWriter(sinkPath, fileSet), P.Effect.provideService(FuncSmithContextFs, funcSmithContextFs)),
    })
  )
);

export const FuncSmithContextWriterLive = P.pipe(FuncSmithContextWriterLayer, P.Layer.provide(FuncSmithContextFsLive));
export const FuncSmithContextWriterTest = P.pipe(FuncSmithContextWriterLayer, P.Layer.provide(FuncSmithContextFsTest));

// --------------------------------------------------------------------------
export const FuncSmithContextEnvDefault = P.Layer.succeed(
  FuncSmithContextEnv,
  FuncSmithContextEnv.of({
    env: {},
  })
);

// --------------------------------------------------------------------------
export const FuncSmithContextMetadataDefault = P.Layer.succeed(
  FuncSmithContextMetadata,
  FuncSmithContextMetadata.of({
    metadata: {},
  })
);

export const FuncSmithContextMetadataTest = P.Layer.succeed(
  FuncSmithContextMetadata,
  FuncSmithContextMetadata.of({
    metadata: {
      testMetaDataName1: 'testMetaDataValue1',
    },
  })
);

// --------------------------------------------------------------------------
export const FuncSmithContextTest = P.Layer.succeed(
  FuncSmithContext(),
  FuncSmithContext().of({
    rootDirPath: '/tmp',
    fileSet: EMPTY_FILESET(),
  })
);
