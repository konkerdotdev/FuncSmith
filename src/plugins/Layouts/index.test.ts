import * as P from '@konker.dev/effect-ts-prelude';

import { FsDepContextTest, FsDepEnvDefault, FsDepMetadataDefault, FsDepReaderTest } from '../../layers';
import * as fixturesFsFm from '../../test/fixtures/fileset-frontmatter-1';
import * as unit from './index';

describe('plugins', () => {
  describe('Layouts', () => {
    it('should return the expected value with default options', async () => {
      const pluginStack = P.pipe(P.Effect.succeed, unit.layouts());
      const actual = await P.Effect.runPromise(
        P.pipe(
          [...fixturesFsFm.TEST_FILE_SET_FRONT_MATTER_1],
          pluginStack,
          P.Effect.provide(FsDepReaderTest),
          P.Effect.provide(FsDepEnvDefault),
          P.Effect.provide(FsDepMetadataDefault),
          P.Effect.provide(FsDepContextTest)
        )
      );
      expect(actual).toMatchSnapshot('plugin-layouts-1');
    });

    it('should return the expected value with specific options, absolute layouts path', async () => {
      const pluginStack = P.pipe(
        P.Effect.succeed,
        unit.layouts({
          layoutsPath: '/tmp/layouts',
        })
      );
      const actual = await P.Effect.runPromise(
        P.pipe(
          [...fixturesFsFm.TEST_FILE_SET_FRONT_MATTER_1],
          pluginStack,
          P.Effect.provide(FsDepReaderTest),
          P.Effect.provide(FsDepEnvDefault),
          P.Effect.provide(FsDepMetadataDefault),
          P.Effect.provide(FsDepContextTest)
        )
      );
      expect(actual).toMatchSnapshot('plugin-layouts-1');
    });

    it('should return the expected value with specific options relative partials path', async () => {
      const pluginStack = P.pipe(
        P.Effect.succeed,
        unit.layouts({
          partialsPath: 'partials',
        })
      );
      const actual = await P.Effect.runPromise(
        P.pipe(
          [...fixturesFsFm.TEST_FILE_SET_FRONT_MATTER_1],
          pluginStack,
          P.Effect.provide(FsDepReaderTest),
          P.Effect.provide(FsDepEnvDefault),
          P.Effect.provide(FsDepMetadataDefault),
          P.Effect.provide(FsDepContextTest)
        )
      );
      expect(actual).toMatchSnapshot('plugin-layouts-1');
    });

    it('should return the expected value with specific options absolute partials path', async () => {
      const pluginStack = P.pipe(
        P.Effect.succeed,
        unit.layouts({
          layoutsPath: '/tmp/layouts',
          partialsPath: '/tmp/partials',
        })
      );
      const actual = await P.Effect.runPromise(
        P.pipe(
          [...fixturesFsFm.TEST_FILE_SET_FRONT_MATTER_1],
          pluginStack,
          P.Effect.provide(FsDepReaderTest),
          P.Effect.provide(FsDepEnvDefault),
          P.Effect.provide(FsDepMetadataDefault),
          P.Effect.provide(FsDepContextTest)
        )
      );
      expect(actual).toMatchSnapshot('plugin-layouts-1');
    });
  });
});
