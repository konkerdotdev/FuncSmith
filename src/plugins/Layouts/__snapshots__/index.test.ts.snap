// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugins Layouts should return the expected value with default options: plugin-layouts-1 1`] = `
[
  {
    "_id": "0000000000001111111111112222222222220001",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": Uint8Array [
      65,
    ],
    "fileBase": "a",
    "fileExt": ".txt",
    "fileName": "a.txt",
    "link": "/a.txt",
    "path": "/tmp/foo/a.txt",
    "relDir": ".",
    "relPath": "a.txt",
  },
  {
    "_id": "0000000000001111111111112222222222220002",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "FOO: P1",
    "date": "2024-01-01",
    "draft": false,
    "fileBase": "p1",
    "fileExt": ".md",
    "fileName": "p1.md",
    "frontMatter": {
      "date": "2024-01-01",
      "draft": false,
      "layout": "foo.hbs",
      "title": "P1",
    },
    "layout": "foo.hbs",
    "link": "/posts/p1.md",
    "path": "/tmp/foo/posts/p1.md",
    "relDir": "posts",
    "relPath": "posts/p1.md",
    "title": "P1",
  },
  {
    "_id": "0000000000001111111111112222222222220003",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "BAR: P2",
    "date": "2024-02-02",
    "draft": true,
    "fileBase": "p2",
    "fileExt": ".md",
    "fileName": "p2.md",
    "frontMatter": {
      "date": "2024-02-02",
      "draft": true,
      "layout": "bar.hbs",
      "title": "P2",
    },
    "layout": "bar.hbs",
    "link": "/posts/p2.md",
    "path": "/tmp/foo/posts/p2.md",
    "relDir": "posts",
    "relPath": "posts/p2.md",
    "title": "P2",
  },
  {
    "_id": "0000000000001111111111112222222222220004",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "LAYOUT P3",
    "date": "2024-03-03",
    "fileBase": "p3",
    "fileExt": ".md",
    "fileName": "p3.md",
    "frontMatter": {
      "date": "2024-03-03",
      "title": "P3",
    },
    "link": "/posts/p3.md",
    "path": "/tmp/foo/posts/p3.md",
    "relDir": "posts",
    "relPath": "posts/p3.md",
    "title": "P3",
  },
  {
    "_id": "0000000000001111111111112222222222220005",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "LAYOUT P4",
    "date": "2024-04-04",
    "fileBase": "p4",
    "fileExt": ".md",
    "fileName": "p4.md",
    "frontMatter": {
      "date": "2024-04-04",
      "title": "P4",
    },
    "link": "/posts/p4.md",
    "path": "/tmp/foo/posts/p4.md",
    "relDir": "posts",
    "relPath": "posts/p4.md",
    "title": "P4",
  },
  {
    "_id": "0000000000001111111111112222222222220006",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "FOO: D1",
    "date": "2024-11-01",
    "fileBase": "p1",
    "fileExt": ".doc",
    "fileName": "d1.doc",
    "frontMatter": {
      "date": "2024-11-01",
      "layout": "foo.hbs",
      "title": "D1",
    },
    "layout": "foo.hbs",
    "link": "/docs/d1.doc",
    "path": "/tmp/foo/docs/d1.md",
    "relDir": "docs",
    "relPath": "docs/d1.doc",
    "title": "D1",
  },
  {
    "_id": "0000000000001111111111112222222222220007",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "BAR: D2",
    "date": "2024-11-02",
    "fileBase": "d2",
    "fileExt": ".doc",
    "fileName": "d2.doc",
    "frontMatter": {
      "date": "2024-11-02",
      "layout": "bar.hbs",
      "title": "D2",
    },
    "layout": "bar.hbs",
    "link": "/docs/d2.doc",
    "path": "/tmp/foo/docs/d2.doc",
    "relDir": "docs",
    "relPath": "docs/d2.doc",
    "title": "D2",
  },
  {
    "_id": "0000000000001111111111112222222222220008",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "LAYOUT P3",
    "date": "2024-11-03",
    "fileBase": "d3",
    "fileExt": ".doc",
    "fileName": "d3.doc",
    "frontMatter": {
      "date": "2024-03-03",
      "title": "P3",
    },
    "link": "/docs/d3.doc",
    "path": "/tmp/foo/docs/d3.doc",
    "relDir": "docs",
    "relPath": "docs/d3.doc",
    "title": "D3",
  },
]
`;

exports[`plugins Layouts should return the expected value with specific options absolute partials path: plugin-layouts-1 1`] = `
[
  {
    "_id": "0000000000001111111111112222222222220001",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": Uint8Array [
      65,
    ],
    "fileBase": "a",
    "fileExt": ".txt",
    "fileName": "a.txt",
    "link": "/a.txt",
    "path": "/tmp/foo/a.txt",
    "relDir": ".",
    "relPath": "a.txt",
  },
  {
    "_id": "0000000000001111111111112222222222220002",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "FOO: P1",
    "date": "2024-01-01",
    "draft": false,
    "fileBase": "p1",
    "fileExt": ".md",
    "fileName": "p1.md",
    "frontMatter": {
      "date": "2024-01-01",
      "draft": false,
      "layout": "foo.hbs",
      "title": "P1",
    },
    "layout": "foo.hbs",
    "link": "/posts/p1.md",
    "path": "/tmp/foo/posts/p1.md",
    "relDir": "posts",
    "relPath": "posts/p1.md",
    "title": "P1",
  },
  {
    "_id": "0000000000001111111111112222222222220003",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "BAR: P2",
    "date": "2024-02-02",
    "draft": true,
    "fileBase": "p2",
    "fileExt": ".md",
    "fileName": "p2.md",
    "frontMatter": {
      "date": "2024-02-02",
      "draft": true,
      "layout": "bar.hbs",
      "title": "P2",
    },
    "layout": "bar.hbs",
    "link": "/posts/p2.md",
    "path": "/tmp/foo/posts/p2.md",
    "relDir": "posts",
    "relPath": "posts/p2.md",
    "title": "P2",
  },
  {
    "_id": "0000000000001111111111112222222222220004",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "LAYOUT P3",
    "date": "2024-03-03",
    "fileBase": "p3",
    "fileExt": ".md",
    "fileName": "p3.md",
    "frontMatter": {
      "date": "2024-03-03",
      "title": "P3",
    },
    "link": "/posts/p3.md",
    "path": "/tmp/foo/posts/p3.md",
    "relDir": "posts",
    "relPath": "posts/p3.md",
    "title": "P3",
  },
  {
    "_id": "0000000000001111111111112222222222220005",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "LAYOUT P4",
    "date": "2024-04-04",
    "fileBase": "p4",
    "fileExt": ".md",
    "fileName": "p4.md",
    "frontMatter": {
      "date": "2024-04-04",
      "title": "P4",
    },
    "link": "/posts/p4.md",
    "path": "/tmp/foo/posts/p4.md",
    "relDir": "posts",
    "relPath": "posts/p4.md",
    "title": "P4",
  },
  {
    "_id": "0000000000001111111111112222222222220006",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "FOO: D1",
    "date": "2024-11-01",
    "fileBase": "p1",
    "fileExt": ".doc",
    "fileName": "d1.doc",
    "frontMatter": {
      "date": "2024-11-01",
      "layout": "foo.hbs",
      "title": "D1",
    },
    "layout": "foo.hbs",
    "link": "/docs/d1.doc",
    "path": "/tmp/foo/docs/d1.md",
    "relDir": "docs",
    "relPath": "docs/d1.doc",
    "title": "D1",
  },
  {
    "_id": "0000000000001111111111112222222222220007",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "BAR: D2",
    "date": "2024-11-02",
    "fileBase": "d2",
    "fileExt": ".doc",
    "fileName": "d2.doc",
    "frontMatter": {
      "date": "2024-11-02",
      "layout": "bar.hbs",
      "title": "D2",
    },
    "layout": "bar.hbs",
    "link": "/docs/d2.doc",
    "path": "/tmp/foo/docs/d2.doc",
    "relDir": "docs",
    "relPath": "docs/d2.doc",
    "title": "D2",
  },
  {
    "_id": "0000000000001111111111112222222222220008",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "LAYOUT P3",
    "date": "2024-11-03",
    "fileBase": "d3",
    "fileExt": ".doc",
    "fileName": "d3.doc",
    "frontMatter": {
      "date": "2024-03-03",
      "title": "P3",
    },
    "link": "/docs/d3.doc",
    "path": "/tmp/foo/docs/d3.doc",
    "relDir": "docs",
    "relPath": "docs/d3.doc",
    "title": "D3",
  },
]
`;

exports[`plugins Layouts should return the expected value with specific options relative partials path: plugin-layouts-1 1`] = `
[
  {
    "_id": "0000000000001111111111112222222222220001",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": Uint8Array [
      65,
    ],
    "fileBase": "a",
    "fileExt": ".txt",
    "fileName": "a.txt",
    "link": "/a.txt",
    "path": "/tmp/foo/a.txt",
    "relDir": ".",
    "relPath": "a.txt",
  },
  {
    "_id": "0000000000001111111111112222222222220002",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "FOO: P1",
    "date": "2024-01-01",
    "draft": false,
    "fileBase": "p1",
    "fileExt": ".md",
    "fileName": "p1.md",
    "frontMatter": {
      "date": "2024-01-01",
      "draft": false,
      "layout": "foo.hbs",
      "title": "P1",
    },
    "layout": "foo.hbs",
    "link": "/posts/p1.md",
    "path": "/tmp/foo/posts/p1.md",
    "relDir": "posts",
    "relPath": "posts/p1.md",
    "title": "P1",
  },
  {
    "_id": "0000000000001111111111112222222222220003",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "BAR: P2",
    "date": "2024-02-02",
    "draft": true,
    "fileBase": "p2",
    "fileExt": ".md",
    "fileName": "p2.md",
    "frontMatter": {
      "date": "2024-02-02",
      "draft": true,
      "layout": "bar.hbs",
      "title": "P2",
    },
    "layout": "bar.hbs",
    "link": "/posts/p2.md",
    "path": "/tmp/foo/posts/p2.md",
    "relDir": "posts",
    "relPath": "posts/p2.md",
    "title": "P2",
  },
  {
    "_id": "0000000000001111111111112222222222220004",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "LAYOUT P3",
    "date": "2024-03-03",
    "fileBase": "p3",
    "fileExt": ".md",
    "fileName": "p3.md",
    "frontMatter": {
      "date": "2024-03-03",
      "title": "P3",
    },
    "link": "/posts/p3.md",
    "path": "/tmp/foo/posts/p3.md",
    "relDir": "posts",
    "relPath": "posts/p3.md",
    "title": "P3",
  },
  {
    "_id": "0000000000001111111111112222222222220005",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "LAYOUT P4",
    "date": "2024-04-04",
    "fileBase": "p4",
    "fileExt": ".md",
    "fileName": "p4.md",
    "frontMatter": {
      "date": "2024-04-04",
      "title": "P4",
    },
    "link": "/posts/p4.md",
    "path": "/tmp/foo/posts/p4.md",
    "relDir": "posts",
    "relPath": "posts/p4.md",
    "title": "P4",
  },
  {
    "_id": "0000000000001111111111112222222222220006",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "FOO: D1",
    "date": "2024-11-01",
    "fileBase": "p1",
    "fileExt": ".doc",
    "fileName": "d1.doc",
    "frontMatter": {
      "date": "2024-11-01",
      "layout": "foo.hbs",
      "title": "D1",
    },
    "layout": "foo.hbs",
    "link": "/docs/d1.doc",
    "path": "/tmp/foo/docs/d1.md",
    "relDir": "docs",
    "relPath": "docs/d1.doc",
    "title": "D1",
  },
  {
    "_id": "0000000000001111111111112222222222220007",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "BAR: D2",
    "date": "2024-11-02",
    "fileBase": "d2",
    "fileExt": ".doc",
    "fileName": "d2.doc",
    "frontMatter": {
      "date": "2024-11-02",
      "layout": "bar.hbs",
      "title": "D2",
    },
    "layout": "bar.hbs",
    "link": "/docs/d2.doc",
    "path": "/tmp/foo/docs/d2.doc",
    "relDir": "docs",
    "relPath": "docs/d2.doc",
    "title": "D2",
  },
  {
    "_id": "0000000000001111111111112222222222220008",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "LAYOUT P3",
    "date": "2024-11-03",
    "fileBase": "d3",
    "fileExt": ".doc",
    "fileName": "d3.doc",
    "frontMatter": {
      "date": "2024-03-03",
      "title": "P3",
    },
    "link": "/docs/d3.doc",
    "path": "/tmp/foo/docs/d3.doc",
    "relDir": "docs",
    "relPath": "docs/d3.doc",
    "title": "D3",
  },
]
`;

exports[`plugins Layouts should return the expected value with specific options, absolute layouts path: plugin-layouts-1 1`] = `
[
  {
    "_id": "0000000000001111111111112222222222220001",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": Uint8Array [
      65,
    ],
    "fileBase": "a",
    "fileExt": ".txt",
    "fileName": "a.txt",
    "link": "/a.txt",
    "path": "/tmp/foo/a.txt",
    "relDir": ".",
    "relPath": "a.txt",
  },
  {
    "_id": "0000000000001111111111112222222222220002",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "FOO: P1",
    "date": "2024-01-01",
    "draft": false,
    "fileBase": "p1",
    "fileExt": ".md",
    "fileName": "p1.md",
    "frontMatter": {
      "date": "2024-01-01",
      "draft": false,
      "layout": "foo.hbs",
      "title": "P1",
    },
    "layout": "foo.hbs",
    "link": "/posts/p1.md",
    "path": "/tmp/foo/posts/p1.md",
    "relDir": "posts",
    "relPath": "posts/p1.md",
    "title": "P1",
  },
  {
    "_id": "0000000000001111111111112222222222220003",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "BAR: P2",
    "date": "2024-02-02",
    "draft": true,
    "fileBase": "p2",
    "fileExt": ".md",
    "fileName": "p2.md",
    "frontMatter": {
      "date": "2024-02-02",
      "draft": true,
      "layout": "bar.hbs",
      "title": "P2",
    },
    "layout": "bar.hbs",
    "link": "/posts/p2.md",
    "path": "/tmp/foo/posts/p2.md",
    "relDir": "posts",
    "relPath": "posts/p2.md",
    "title": "P2",
  },
  {
    "_id": "0000000000001111111111112222222222220004",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "LAYOUT P3",
    "date": "2024-03-03",
    "fileBase": "p3",
    "fileExt": ".md",
    "fileName": "p3.md",
    "frontMatter": {
      "date": "2024-03-03",
      "title": "P3",
    },
    "link": "/posts/p3.md",
    "path": "/tmp/foo/posts/p3.md",
    "relDir": "posts",
    "relPath": "posts/p3.md",
    "title": "P3",
  },
  {
    "_id": "0000000000001111111111112222222222220005",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "LAYOUT P4",
    "date": "2024-04-04",
    "fileBase": "p4",
    "fileExt": ".md",
    "fileName": "p4.md",
    "frontMatter": {
      "date": "2024-04-04",
      "title": "P4",
    },
    "link": "/posts/p4.md",
    "path": "/tmp/foo/posts/p4.md",
    "relDir": "posts",
    "relPath": "posts/p4.md",
    "title": "P4",
  },
  {
    "_id": "0000000000001111111111112222222222220006",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "FOO: D1",
    "date": "2024-11-01",
    "fileBase": "p1",
    "fileExt": ".doc",
    "fileName": "d1.doc",
    "frontMatter": {
      "date": "2024-11-01",
      "layout": "foo.hbs",
      "title": "D1",
    },
    "layout": "foo.hbs",
    "link": "/docs/d1.doc",
    "path": "/tmp/foo/docs/d1.md",
    "relDir": "docs",
    "relPath": "docs/d1.doc",
    "title": "D1",
  },
  {
    "_id": "0000000000001111111111112222222222220007",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "BAR: D2",
    "date": "2024-11-02",
    "fileBase": "d2",
    "fileExt": ".doc",
    "fileName": "d2.doc",
    "frontMatter": {
      "date": "2024-11-02",
      "layout": "bar.hbs",
      "title": "D2",
    },
    "layout": "bar.hbs",
    "link": "/docs/d2.doc",
    "path": "/tmp/foo/docs/d2.doc",
    "relDir": "docs",
    "relPath": "docs/d2.doc",
    "title": "D2",
  },
  {
    "_id": "0000000000001111111111112222222222220008",
    "_tag": "File",
    "baseDir": "/tmp/foo",
    "contents": "LAYOUT P3",
    "date": "2024-11-03",
    "fileBase": "d3",
    "fileExt": ".doc",
    "fileName": "d3.doc",
    "frontMatter": {
      "date": "2024-03-03",
      "title": "P3",
    },
    "link": "/docs/d3.doc",
    "path": "/tmp/foo/docs/d3.doc",
    "relDir": "docs",
    "relPath": "docs/d3.doc",
    "title": "D3",
  },
]
`;
